okx/xlayer-erigon
# Truncate data stream while launching the Node

**Function:** 
- Allow truncation of data stream blocks via configuration file.
- Halted the node due to a serious error while writing the data stream 

**Case:** 
- In extreme situations where the data stream is corrupted or lost at a certain block height, Erigon does not have an automatic rollback feature. 
- By configuring the truncate block height and restarting, new data streams can be quickly regenerated.

**Test Result:**
- Configure `zkevm.data-stream-truncate-block`any height and restart the sequencer to truncate and regenerate data stream.
- Delete the RPC node's data completely, restart again and the syncing work as expected.
